tools/rewriter/GenCallAsm.cpp

```diff
@@ -683,15 +683,31 @@ static void emit_copy_args(AsmWriter &aw, const std::vector<ArgLocation> &args,
   }
 }
 
-static void emit_set_pkru(AsmWriter &aw, uint32_t target_pkey, Arch arch) {
+static void emit_set_pkru(AsmWriter &aw, uint32_t target_pkey, Arch arch, std::optional<uint32_t> union_pkey = std::nullopt) {
   if (arch == Arch::X86) {
-    // Change pkru to the compartment's value
-    add_comment_line(aw, "Set PKRU to the compartment's value");
+#ifdef IA2_LIBC_COMPARTMENT
```

Comment:
nit: it'd be nice to turn this compile-time flag into a run-time flag so there's only one build on the rewriter. You can use the `Arch` enum as a guide on how to do this.